src/syscall/syscall_libc.go

```diff
@@ -68,6 +69,86 @@ func Rmdir(path string) (err error) {
 	return
 }
 
+type Timespec struct {
+	Sec  int64
+	Nsec int64
+}
+
+// Go chose Linux's field names for Stat_t, see https://github.com/golang/go/issues/31735
+type Stat_t struct {
+	Dev       int32
+	Mode      uint16
+	Nlink     uint16
+	Ino       uint64
+	Uid       uint32
+	Gid       uint32
+	Rdev      int32
+	Pad_cgo_0 [4]byte
+	Atim      Timespec
+	Mtim      Timespec
+	Ctim      Timespec
+	Btim      Timespec
+	Size      int64
+	Blocks    int64
+	Blksize   int32
+	Flags     uint32
+	Gen       uint32
+	Lspare    int32
+	Qspare    [2]int64
+}
+
+// Source: https://github.com/apple/darwin-xnu/blob/main/bsd/sys/_types/_s_ifmt.h
+// Consistent with values we use from lib/wasi-libc/sysroot/include/sys/stat.h
```

Comment:
Even with that, I'd prefer if this lived in syscall_libc_darwin.go in the (unlikely) case that a different OS uses different constants. I would prefer a little copying over deduplication in this case, unless these constants are some sort of standard (which they don't appear to be).

(Also, sidenote: this syscall package is not used on Linux. We use the upstream Go syscall package on Linux).

Reply:
Thanks.  I had it in syscall_libc_darwin originally, but moved it to get nintendo switch to pass, which was probably dumb :-)

Reply:
Sorted